src/pages/billing/models/BillingProject.ts

```diff
@@ -0,0 +1,26 @@
+import { AzureManagedAppCoordinates } from 'src/pages/billing/models/AzureManagedAppCoordinates'
+
+
+export type CloudPlatform = 'GCP' | 'Azure'
```

Comment:
In the server response, it's actually 'AZURE'.

```
{"cloudPlatform":"AZURE","invalidBillingAccount":false,"managedAppCoordinates":{"managedResourceGroupId":"mrg-terra-azure-dev-20221202091509","subscriptionId":"3efc5bdf-be0e-44e7-b1d7-c08931e3c16c","tenantId":"0cb7a640-45a2-4ed6-be9f-63519f86e04b"},"projectName":"azure-dev-2022-12-02","roles":["Owner"],"status":"Ready"},{"cloudPlatform":"AZURE","invalidBillingAccount":false,"managedAppCoordinates":{"managedResourceGroupId":"mrg-terra-dev-jan23-20230123125907","subscriptionId":"3efc5bdf-be0e-44e7-b1d7-c08931e3c16c","tenantId":"0cb7a640-45a2-4ed6-be9f-63519f86e04b"},"projectName":"azure-dev-2023-01-23","roles":["Owner"],"status":"Ready"},{"billingAccount":"billingAccounts/01A82E-CA8A14-367457","cloudPlatform":"GCP","invalidBillingAccount":false,"projectName":"callisto-dev","roles":["User","Owner"],"status":"Ready"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: b23e9f9a-a39e-472d-be48-46a4d1c337e5","projectName":"CARAzureRawls","roles":["Owner"],"status":"Error"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: acdb4ac1-9bd3-4078-a686-6380f2b011b0","projectName":"CARAzureTake3","roles":["Owner"],"status":"Error"},{"billingAccount":"billingAccounts/00708C-45D19D-27AAFA","cloudPlatform":"GCP","invalidBillingAccount":false,"projectName":"CARBilling-2","roles":["Owner"],"status":"Ready"},{"cloudPlatform":"GCP","invalidBillingAccount":false,"projectName":"CARBilling3","roles":["Owner"],"status":"Ready"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: bf45a290-4da9-4dde-a939-1e5e414c4725","projectName":"CARBilling4","roles":["Owner"],"status":"Error"},{"billingAccount":"billingAccounts/00708C-45D19D-27AAFA","cloudPlatform":"GCP","invalidBillingAccount":true,"projectName":"CARBillingTest","roles":["Owner"],"status":"Ready"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: c2ab9efd-8204-403d-84b1-4f5edd906785","projectName":"CARDebugLZ","roles":["Owner"],"status":"Error"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: 9504473f-561a-46e5-8093-d209a7f1bdc0","projectName":"CARDebugLZ2","roles":["Owner"],"status":"Error"},{"cloudPlatform":"AZURE","invalidBillingAccount":false,"managedAppCoordinates":{"managedResourceGroupId":"mrg-terra-dev-previ-20221213131637","subscriptionId":"df547342-9cfd-44ef-a6dd-df0ede32f1e3","tenantId":"fad90753-2022-4456-9b0a-c7e5b934e408"},"projectName":"CARDec13Work","roles":["Owner"],"status":"Ready"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: 6b01b991-5a7a-4328-8961-1654057beddb","projectName":"CARDec22Azure","roles":["Owner"],"status":"Error"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: de966e9a-59d4-47ed-9610-8df7e59b16b9","projectName":"CARDec22Take2","roles":["Owner"],"status":"Error"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: 515abd5d-8887-4c7e-88ab-5419d534cf4a","projectName":"CARDec7WS","roles":["Owner"],"status":"Error"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: bb030783-7dd7-4012-8960-61386191026f","projectName":"CARDec9WOR","roles":["Owner"],"status":"Error"},{"cloudPlatform":"AZURE","invalidBillingAccount":false,"managedAppCoordinates":{"managedResourceGroupId":"mrg-terra-dev-previ-20230119110658","subscriptionId":"df547342-9cfd-44ef-a6dd-df0ede32f1e3","tenantId":"fad90753-2022-4456-9b0a-c7e5b934e408"},"projectName":"CARJan19bemis","roles":["Owner"],"status":"Ready"},{"cloudPlatform":"AZURE","invalidBillingAccount":false,"managedAppCoordinates":{"managedResourceGroupId":"mrg-terra-dev-previ-20230131123105","subscriptionId":"df547342-9cfd-44ef-a6dd-df0ede32f1e3","tenantId":"fad90753-2022-4456-9b0a-c7e5b934e408"},"projectName":"CARJan31AzureWest","roles":["Owner"],"status":"Ready"},{"cloudPlatform":"AZURE","invalidBillingAccount":false,"managedAppCoordinates":{"managedResourceGroupId":"mrg-terra-dev-previ-20221108090449","subscriptionId":"62b22893-6bc1-46d9-8a90-806bb3cce3c9","tenantId":"fad90753-2022-4456-9b0a-c7e5b934e408"},"projectName":"CARTerraNov8_ws1","roles":["Owner"],"status":"Ready"},{"billingAccount":"billingAccounts/01A82E-CA8A14-367457","cloudPlatform":"GCP","invalidBillingAccount":false,"projectName":"galaxy-anvil-dev","roles":["Owner"],"status":"Ready"},{"billingAccount":"billingAccounts/00708C-45D19D-27AAFA","cloudPlatform":"GCP","invalidBillingAccount":false,"projectName":"general-dev-billing-account","roles":["User"],"status":"Ready"},{"cloudPlatform":"AZURE","invalidBillingAccount":false,"managedAppCoordinates":{"managedResourceGroupId":"mrg-terra-dev-previ-20220930130036","subscriptionId":"3efc5bdf-be0e-44e7-b1d7-c08931e3c16c","tenantId":"0cb7a640-45a2-4ed6-be9f-63519f86e04b"},"projectName":"terra-dev-test-project","roles":["Owner"],"status":"Ready"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: 90282f4f-af84-479b-9472-02dc8533d510","projectName":"WorkspacesDemo","roles":["Owner"],"status":"Error"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: e140fc23-206f-428c-9978-468f854476b8","projectName":"WorkspacesDemo2","roles":["Owner"],"status":"Error"},{"cloudPlatform":"UNKNOWN","invalidBillingAccount":false,"message":"Unable to find billing profile in Billing Profile Manager for billing profile id: 651011ae-8006-45e2-91a3-fcc5adf66551","projectName":"WorkspacesDemo3","roles":["Owner"],"status":"Error"}]

Reply:
'UNKNOWN' is also a possible value, when we can't retrieve the record from BPM. It's an error case, but Rawls does document it and I have a billing project in that state.

Reply:
Good point: I didn't really think about the unknown case.